pkg/broker/filter/filter_handler.go

```diff
@@ -206,34 +219,51 @@ func (h *Handler) ServeHTTP(writer http.ResponseWriter, request *http.Request) {
 
 	h.reportArrivalTime(event, reportArgs)
 
-	h.send(ctx, writer, request.Header, subscriberURI.String(), reportArgs, event, ttl)
+	h.send(ctx, writer, request.Header, subscriberURI.URL(), reportArgs, event, ttl)
 }
 
-func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers http.Header, target string, reportArgs *ReportArgs, event *cloudevents.Event, ttl int32) {
+func (h *Handler) send(ctx context.Context, writer http.ResponseWriter, headers http.Header, target *url.URL, reportArgs *ReportArgs, event *cloudevents.Event, ttl int32) {
 	// send the event to trigger's subscriber
-	response, err := h.sendEvent(ctx, headers, target, event, reportArgs)
+	dispatchInfo, response, err := h.sendEvent(ctx, headers, target, event, reportArgs)
+
 	if err != nil {
 		h.logger.Error("failed to send event", zap.Error(err))
-		writer.WriteHeader(http.StatusInternalServerError)
-		_ = h.reporter.ReportEventCount(reportArgs, http.StatusInternalServerError)
+
+		writer.WriteHeader(dispatchInfo.ResponseCode)
```

Comment:
If the error is caused by a different problem and we couldn't get the actual response, I'd expect `dispatchInfo` to be nil and we should respond with  `http.StatusInternalServerError`

Reply:
This is one case, for example (line 260):
```
	req, err := h.sender.NewCloudEventRequestWithTarget(ctx, target.String())
	if err != nil {
		return nil, fmt.Errorf("failed to create the request: %w", err)
		return nil, nil, fmt.Errorf("failed to create the request: %w", err)
	}
```

Reply:
Other thought, the returned `DispatchInfo` is actually used only in the error path, can we create a custom error to propagate those info to the caller?

Reply:
Got yout point:)